### ISSUE TITLE
The `far` member of `PerspectiveProjection` is irrelevant

### ISSUE THREAD

**Issue by HackerFoo** [bevyengine/bevy#4596]
The `far` member of `PerspectiveProjection` is irrelevant, because the far plane is at negative infinite Z

https://github.com/bevyengine/bevy/blob/989fb8a78d2ca6e3fe6fb1ccd7d79ea88868b8c7/crates/bevy_render/src/camera/projection.rs#L25

**Comment by mockersf** [comment]
maybe related to https://github.com/bevyengine/bevy/pull/4457

**Comment by superdump** [comment]
We could still cull meshes to a fake far plane. I just chose not to to preserve behaviour at the time I think. I’m not sure whether we should or shouldn’t cull to a fake far plane. We do for point lights but not for meshes. We could make it optional but then I don’t know what the default should be. I kind of feel like we could make it an Option<f32> and if present then we cull to it, if not, we don’t.

**Comment by cart** [comment]
Yeah I think providing optional configuration for people to cull at a given distance makes a lot of sense.

**Comment by superdump** [comment]
- I've had a go at this in #5282
(The above is a bullet point because that's the only way I know to force github markdown to show the linked PR title.)

**Comment by Neopallium** [comment]
I have been thinking of culling smaller objects faster than large objects.  This way large buildings would still be visible, but smaller objects inside or around it wouldn't be rendered.

Would it be difficult to do that in the far-plane culling logic?

**Comment by superdump** [comment]
What do you mean by faster? All meshes are chilled against the visible volume every frame.

**Comment by Neopallium** [comment]
Sorry, by "faster" I mean at a closer distance.  A small object (scale dimensions < 1.0) will not be as visible (size on screen) at a distance of 50.0 from the camera vs a larger object (scale dim. > 1.0).

I have a 3d scene that has +40,000 objects (over 100,000 ECS entities).  The object meshes are generated from primitive profile/path parameters and can have different textures on different faces.   So right now one object could have multiple meshes.

I had tried enabling the far plane culling like in your PR to see if it would help with FPS.  It does, but there is a hard cut-off, where even large objects go invisible.

I am now using an adaptive culling based on distance and the Transform.scale of the mesh.  For the scene that I have tested with (large buildings with many smaller objects inside), it allows the buildings that are far away to still be visible while hiding the smaller objects that are not as noticable (even when the build has lots of windows).
